javalib/src/main/scala/java/util/AbstractCollection.scala

```diff
@@ -0,0 +1,89 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.immutable.{ List => SList }
+
+abstract class AbstractCollection[E]() extends java.util.Collection[E] {
+  def iterator(): Iterator[E]
+  def size(): Int
+
+  @tailrec
+  private def iterateAndDo[X](i: Iterator[E], f: (E) => X, part: SList[X] = SList()): SList[X] =
+    if (i.hasNext) iterateAndDo(i, f, f(i.next) :: part)
+    else part
+
+  @tailrec
+  private def iterateApplyAndDo[X](i: Iterator[E], f: (Iterator[E], E) => X, part: SList[X] = SList()): SList[X] =
+    if (i.hasNext) iterateApplyAndDo(i, f, f(i, i.next) :: part)
+    else part
+
+  @tailrec
+  private def iterateAndCheck[X](i: Iterator[X], f: (X) => Boolean): Boolean =
+    if (i.hasNext) {
+      if (f(i.next)) true
+      else iterateAndCheck(i, f)
+    } else false
```

Comment:
Add `{}` around the body of the method, because it is neither a one-liner, nor a 2-liners `if/else`.